ironfish/src/rpc/adapters/tcpAdapter.ts

```diff
@@ -110,33 +117,45 @@ export class TcpAdapter implements IAdapter {
   }
 
   onClientConnection(socket: net.Socket): void {
-    const connId = uuid()
-    const reqMap = new Map<string, Request>()
-    this.pending.set(connId, { sock: socket, reqs: reqMap })
+    const requests = new Map<string, Request>()
+    const client = { socket, requests, id: uuid() }
+    this.clients.set(client.id, client)
+
     socket.on('data', (data) => {
-      this.onClientData(socket, data, reqMap).catch((err) => this.logger.error(err))
+      this.onClientData(client, data, requests).catch((e) => {
+        this.onClientError(client, e)
+      })
     })
+
     socket.on('close', () => {
-      // When the socket is closed, close all open requests and delete the connection
-      reqMap.forEach((req) => req.close())
-      this.pending.delete(connId)
-      this.logger.debug(`client connection closed: ${this.host}:${this.port}`)
+      this.onClientDisconnection(client)
     })
+
     socket.on('error', (error: Error) => {
-      this.logger.debug(`${this.host}:${this.port} has error : ${error.message}`)
+      this.onClientError(client, error)
     })
   }
 
+  onClientDisconnection(client: TcpAdapterClient): void {
+    client.requests.forEach((req) => req.close())
+    this.clients.delete(client.id)
+    this.logger.debug(`client connection closed: ${this.host}:${this.port}`)
+  }
+
+  onClientError(client: TcpAdapterClient, error: unknown): void {
+    this.logger.debug(`${this.host}:${this.port} has error: ${ErrorUtils.renderError(error)}`)
+  }
+
   async onClientData(
-    socket: net.Socket,
+    client: TcpAdapterClient,
     data: Buffer,
-    reqMap: Map<string, Request>,
+    pending: Map<string, Request>,
```

Comment:
I don't think `pending` is needed here now that the requests are part of the `TcpAdapterClient` instance

Reply:
This code is really hairy, I wasn't sure if I wanted to clean it up in this PR or wait til the next one. I'm down to do it now, as long as you are down to see more code change in this PR.

Reply:
DONE